src/Core/src/ImageSources/Android/ImageLoaderCallback.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Graphics.Drawables;
+
+namespace Microsoft.Maui
+{
+	internal class ImageLoaderCallback : Java.Lang.Object, IImageLoaderCallback
+	{
+		TaskCompletionSource<IImageSourceServiceResult<bool>> tcsResult = new();
+
+		public Task<IImageSourceServiceResult<bool>> Result
+			=> tcsResult.Task;
+
+		public void OnComplete(Java.Lang.Boolean? success, Java.Lang.IRunnable? dispose)
+		{
+			var s = success?.BooleanValue() ?? false;
+
+			Action? disposeWrapper = null;
+			if (dispose != null)
+				disposeWrapper = dispose.Run;
+
+			tcsResult.TrySetResult(new ImageSourceServiceResult(s, disposeWrapper));
+		}
+	}
+
+	internal class ImageLoaderDrawableCallback : Java.Lang.Object, IImageLoaderDrawableCallback
```

Comment:
Is there a way these two could be the same type? You pay a cost per C# type that subclasses Java or per Java type you call from C#.

Reply:
Made these share a type as suggested.